### ISSUE TITLE
emacsclient buffers don't get killed upon C-x #

### ISSUE THREAD

**Issue by kini** [syl20bnr/spacemacs#492]
In plain Emacs, if you run `(server-start)` and then call `emacsclient ~/.bashrc` (for example) from a shell, then `~/.bashrc` will be loaded into a buffer and "When done with a buffer, type `C-x #`" will appear in the minibuffer. In the shell, the `emacsclient` invocation continues to block. Later, when you type `C-x #`, then the buffer is killed, and the `emacsclient` invocation exits back to the shell.

In spacemacs, the beginning works the same, but when you do `C-x #`, the `emacsclient` invocation exits back to the shell but the buffer is not killed. The message "No server buffers remain to edit" appears in the minibuffer.

**Comment by syl20bnr** [comment]
Did you try to reproduce it with version `0.50.0+` ?
What is the value of your `dotspacemacs-persistent-server` variable ?

**Comment by kini** [comment]
Yes, it occurs with version `0.51.0`. `dotspacemacs-persistent-server` is set to `nil`.

**Comment by herbertjones** [comment]
Buffers close as expected when set:  ```(add-hook 'server-done-hook 'kill-buffer)```

**Comment by herbertjones** [comment]
When these lines are removed from config.el, the problem is also fixed.

```
;; keep buffers opened when leaving an emacs client
(setq server-kill-new-buffers nil)
```

Question is, should these lines be removed, or should a special hook be added to maintain the previous behavior?  I'm not sure what the exact reasoning behind these lines were, so I'm not even sure what the hook would be.

**Comment by syl20bnr** [comment]
This is the expected behavior, only the temporary buffers are killed from the server.
To kill all buffers then put this is the dotspacemacs/config:

```elisp
(setq server-kill-new-buffers nil)
```

Thank you @herbertjones for the fix !

**Comment by kini** [comment]
Thanks! But in vanilla Emacs, the `server-kill-new-buffers` variable's default value is `t`. Why does Spacemacs change it to `nil`?

**Comment by syl20bnr** [comment]
Good point, I'm fixing it and move the current config to my dotfile :-)

**Comment by kini** [comment]
Yay, thanks! :)

**Comment by syl20bnr** [comment]
Fixed in 0.64.0
